File: streamer-v2/src/main/java/eu/tsalliance/streamer/listener/TrackEventListener.java
package eu.tsalliance.streamer.listener;

import eu.tsalliance.streamer.channel.AudioTrack;

public interface TrackEventListener {

    int REASON_MAY_START_NEXT = 0;
    int REASON_DONE = 1;
    int REASON_SHUTDOWN = 2;

    /**
     * Fired when AudioTrack started playing
     * @param track AudioTrack
     */
    void onTrackStarted(AudioTrack track);

    /**
     * Fired when AudioTrack stopped playing
     * @param track AudioTrack
     * @param reason Reason constant
     */
    void onTrackEnded(AudioTrack track, int reason);

    /**
     * Fired when an exception occured when playing an AudioTrack
     * @param track AudioTrack where exception occured
     * @param exception Exception that was thrown
     */
    void onTrackException(AudioTrack track, Exception exception);

}

File: nodeserver/src/main/kotlin/live/tsradio/daemon/listener/TrackEventListener.kt
package live.tsradio.daemon.listener

import live.tsradio.daemon.sound.AudioTrack

const val REASON_MAY_START_NEXT = 0
const val REASON_EXCEPTION = 1

interface TrackEventListener {

    fun onTrackStart(track: AudioTrack)
    fun onTrackEnd(track: AudioTrack, endReason: Int, exception: Exception?)

}